### PR TITLE
Feature/1401 own semaphore for sub cache

### PR DESCRIPTION
**Pull request by kzangeli**
Implements #1401 and https://github.com/telefonicaid/fiware-orion/issues/1377

The mongo sub cache is now protected by its own semaphore and now "req mutex policy" con coexist with the subscription cache

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -33,4 +33,5 @@ Add:  Hardening: mongoBackend checking for field existence and proper type to av
 Add:  githash in --version (Issue #1363)
 Add:  Added CLI argument for notification mode. Default mode is 'transient' (previous version used 'permanent' implicitly) (Issue #1370)
 Add:  Added CLI argument for connection memory limit.  Default value is 64Kb (Issue #1384)
+Fix:  Separate semahpore for subscription cache, so that sub cache is independent of mutex policy (Issue #1401)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Probably this is an "aspect" of the #1308 fix, but it doesn't hurt having it as an idepdennt entry, if you prefer so. Up to you (just mark this as NTC if you think is ok as it is now).

Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: As it is a "separate issue", I thought it convenient to mark it here.
What we perhaps should do is to make sure the issue 1401 is clearly marked as "related" to 1308

Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: What about "merging" both line, something like:

```
Fix: Fixed broken subscription cache (Issue #1308), including the semaphore for subscription cache (#1401)
```

Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Sure, no probs

Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Fixed in 72bff47

**Comment by fgalan** [comment]
LGTM

**Pull request merged**